src/librustc/middle/ty.rs

```diff
@@ -749,6 +749,9 @@ pub struct ctxt<'tcx> {
     /// Maps a trait onto a list of impls of that trait.
     pub trait_impls: RefCell<DefIdMap<Rc<RefCell<Vec<ast::DefId>>>>>,
 
+    /// Maps a trait onto a list of negative impls of that trait.
+    pub trait_negative_impls: RefCell<DefIdMap<Rc<RefCell<Vec<ast::DefId>>>>>,
```

Comment:
Originally, I had them separate everywhere and then I started unifying them in a single set after we discussed it on IRC. I did it everywhere but here because it seems useful to be able to access negative implementations from the `tcx` without having to filter `trait_impls` every time.

I can't really think of a case where we *need* to have these separate and we can always filter `trait_impls` if needed. I'll merge these 2.